ambry-store/src/main/java/com.github.ambry.store/DiskManager.java

```diff
@@ -80,13 +88,16 @@
     diskSpaceAllocator = new DiskSpaceAllocator(diskManagerConfig.diskManagerEnableSegmentPooling,
         new File(disk.getMountPath(), diskManagerConfig.diskManagerReserveFileDirName),
         diskManagerConfig.diskManagerRequiredSwapSegmentsPerSize, metrics);
+    this.replicaStatusDelegate = replicaStatusDelegate;
+    this.stoppedReplicas = stoppedReplicas == null ? Collections.emptyList() : stoppedReplicas;
```

Comment:
I think the caller never passes null for `stoppedReplicas` from reading the code. In general, it is best to avoid the same check when the data structures are passed around and instead do it at one place (highest level).

Perhaps a better alternative is to ensure that `ClusterMap.getStoppedReplicas()` - or whichever is the root method from where you get this list - never returns null. Makes for better readability.

Reply:
Changes are made in `ClusterMapUtil.getStoppedReplicas()` to ensure it never returns null.